op-node/sources/types.go

```diff
@@ -83,71 +87,175 @@ func (info *HeaderInfo) ReceiptHash() common.Hash {
 	return info.receiptHash
 }
 
-type rpcHeaderCacheInfo struct {
+type rpcHeader struct {
+	ParentHash  common.Hash      `json:"parentHash"`
+	UncleHash   common.Hash      `json:"sha3Uncles"`
+	Coinbase    common.Address   `json:"miner"`
+	Root        common.Hash      `json:"stateRoot"`
+	TxHash      common.Hash      `json:"transactionsRoot"`
+	ReceiptHash common.Hash      `json:"receiptsRoot"`
+	Bloom       eth.Bytes256     `json:"logsBloom"`
+	Difficulty  hexutil.Big      `json:"difficulty"`
+	Number      hexutil.Uint64   `json:"number"`
+	GasLimit    hexutil.Uint64   `json:"gasLimit"`
+	GasUsed     hexutil.Uint64   `json:"gasUsed"`
+	Time        hexutil.Uint64   `json:"timestamp"`
+	Extra       hexutil.Bytes    `json:"extraData"`
+	MixDigest   common.Hash      `json:"mixHash"`
+	Nonce       types.BlockNonce `json:"nonce"`
+
+	// BaseFee was added by EIP-1559 and is ignored in legacy headers.
+	BaseFee *hexutil.Big `json:"baseFeePerGas" rlp:"optional"`
+
+	// untrusted info included by RPC, may have to be checked
 	Hash common.Hash `json:"hash"`
 }
 
-type rpcHeader struct {
-	cache  rpcHeaderCacheInfo
-	header types.Header
+// checkPostMerge checks that the block header meets all criteria to be a valid ExecutionPayloadHeader,
+// see EIP-3675 (block header changes) and EIP-4399 (mixHash usage for prev-randao)
```

Comment:
These EIPs also mention `ommers` and `ommersHash`. Is that something we need to consider?

Reply:
That's the same as the uncle hash, and is already checked. The EIPs follow yellow paper naming, but "sha3Uncles" is used in the RPC and geth.